util_function_groups/column_operations.R

```diff
@@ -3,6 +3,219 @@
+  starting_w_digit_pattern <- "^\\.?[0-9]"
+  naughty_words <- c("if", "else", "repeat", "while", "function", "for",
+                     "in", "next", "break", "TRUE", "FALSE", "NULL", "Inf",
+                     "NaN", "NA", "NA_integer_", "NA_real_", "NA_complex_", "NA_character_", NA)
+  
+  # if the new names in the mapping match any evil patterns, then complain
+  if( any(mapping$new_name %in% naughty_words) |
+      any(grepl(whitespace_pattern, mapping$new_name)) |
+      any(grepl(starting_w_digit_pattern, mapping$new_name))) {
+    stop("Invalid new names are present among: ",paste(mapping$new_name, collapse =" "))
+  }
+  
+  # if old and new names are not the same length, then complain
+  if(length(mapping$old_name) != length(mapping$new_name)) {
+    stop("Error! Old names and new names are not the same length in your mapping for renaming.")
+  }
+  
+  names(data) <- util.recode(names(data),
+                             mapping$old_name,
+                             mapping$new_name)
+  
+  # if new column names in the final data have duplicates, then complain
+  if (any(duplicated(names(data)))) {
+    stop("Error! Duplicated names in your renamed data. Review your mapping for renaming.")
+  }
+  
+  return(data)
+}
+
+################################ testing for rename_columns  ################################################
+
+######################## TEST 1: handling forbidden names
+
+
+test_forbidden_names <- function() {
+  
+  my_mapping <- list(
+    old_name = c("foo", "bar", "baz","blerp"),
+    new_name = c("newfoo", "newbar", "newbaz", NA)
+  )
+  
+  d <- data.frame(
+    foo = c(1,2,3,4,5),
+    bar = c(6,7,8,9,10),
+    baz = c(11,12,13,14,15),
+    blerp = c(16,17,18,19,20)
+  )
+  
+  success <- FALSE
+  
+  tryCatch(
+    expr = {
+      rename_columns(d,my_mapping)
+    },
+    error = function(e) {
+      success <<- TRUE
+    }
+  )
+  return(success)
+}
+
+test_forbidden_names()
+
+######################## TEST 2: handling extra data columns that aren't in mapping
+
+test_extra_data_columns <- function() {
+  
+  my_mapping <- list(
+    old_name = c("foo"),
+    new_name = c("newfoo")
+  )
+  
+  d <- data.frame(
+    foo = c(1,2,3,4,5),
+    blerp = c(16,17,18,19,20)
+  )
+  
+  expected_output <- data.frame(
+    newfoo = c(1,2,3,4,5),
+    blerp = c(16,17,18,19,20)
+  )
+  
+  new_d <- rename_columns(d,my_mapping)
+  
+  return(identical(new_d, expected_output))
+  
+}
+
+test_extra_data_columns()
+
+######################## TEST 3: extra mapping columns aren't present in data
+
+test_extra_mapping_columns <- function() {
+  
+  my_mapping <- list(
+    old_name = c("foo","bar"),
+    new_name = c("newfoo","newbar")
+  )
+  
+  d <- data.frame(
+    foo = c(1,2,3,4,5)
+  )
+  
+  expected_output <- data.frame(
+    newfoo = c(1,2,3,4,5)
+  )
+  
+  new_d <- rename_columns(d,my_mapping)
+  
+  return(identical(new_d, expected_output))
+  
+}
+
+test_extra_mapping_columns()
```

Comment:
@cmacrander the reason I thought `test_` wasn't good enough was because I was afraid of clearing all objects from the global scope that start with that name pattern.